Sources/Combine/Extensions/Publisher+Utility.swift

```diff
@@ -12,32 +12,148 @@ import Combine
-        var previousValue: Output?
-        return self
-            .prefix(while: { output in
-                guard let previousValue = previousValue else {
-                    return predicate(output)
-                }
-                return predicate(previousValue)
-            })
-            .handleEvents(receiveOutput: {
-                previousValue = $0
-            })
+        return mapToValue(())
+    }
+
+    /// Prefixes elements of a sequence for the given `predicate` while also including the value at which the `prefix` operator stops.
+    func inclusivePrefix(while predicate: @escaping (Self.Output) -> Bool) -> AnyPublisher<Self.Output, Failure> {
+        let shared = share(replay: 1)
+        // In order to get a prefix that's inclusive, we need to make use of the `share(replay:)` operator
+        // First, we prefix as usual until we get all of the elements we want, for the given predicate.
+        // Next, we make use of the replayed sequence to get the first value which doesn't fulfill the predicate requirement.
+        return shared
+            .prefix(while: predicate)
+            .append(shared.first { !predicate($0) })
+            .eraseToAnyPublisher()
+    }
+}
+
+// MARK: Weak assign
+
+@available(iOS 13.0, *)
+public extension Publisher {
+
+    /// Executes specified receiveValue closure if the value has been received, otherwise, if the Publisher
+    /// error'd out, it executes the receiveError closure
+    ///
+    /// - Parameters:
+    ///   - receiveValue: Closure which specifies what should be done if a value has been received
+    ///   - receiveError: Closure which specifies what should be done if an error has been received
+    ///
+    /// - Returns: AnyCancellable
+    func sink(
+        receiveValue: @escaping ((Self.Output) -> Void),
+        receiveError: @escaping ((Self.Failure) -> Void)
+    ) -> AnyCancellable {
+        return sink(
+            receiveCompletion: { result in
+                guard case .failure(let error) = result else { return }
+                receiveError(error)
+            },
+            receiveValue: receiveValue
+        )
+    }
+
+    /// Weakly assigns the received value to the specified object at the specified key path, if the Publisher errors out
+    /// the error will be sent to the specified error handling subject
+    ///
+    /// - Parameters:
+    ///   - object: Object on which you want to assing the value
+    ///   - errorSubject: Subject which will handle the error if it happens
+    ///   - valueKeyPath: Key path on the specified object to which you want to assign the value
+    ///
+    /// - Returns: AnyCancellable
+    func assignWeakifed<T: AnyObject> (
+        on object: T,
+        errorAt errorSubject: PassthroughSubject<Error, Never>? = nil,
+        valueAt valueKeyPath: ReferenceWritableKeyPath<T, Self.Output?>? = nil
```

Comment:
It doesn't make that much sense to have object non-optional, and value optional. They should both be optional or non-optional, since they are both needed to accomplish setting the value.